drivers/hmis/app/graphql/mutations/delete_client_file.rb

```diff
@@ -0,0 +1,23 @@
+module Mutations
+  class DeleteClientFile < BaseMutation
+    argument :file_id, ID, required: true
+
+    field :client, Types::HmisSchema::Client, null: true
+
+    def resolve(file_id:)
+      file = Hmis::File.find_by(id: file_id)
+
+      errors = HmisErrors::Errors.new
+      errors.add :file_id, :not_found unless file.present?
+      errors.add :file_id, :not_allowed if file.present? && !current_user.can_edit_clients_for?(file.client)
```

Comment:
shouldn't this check the file permission? `can_manage_client_files`

Reply:
+1, see suggestion to use common deletion code though